libs/librrgraph/src/io/rr_graph_reader.cpp

```diff
@@ -115,3 +116,58 @@ void load_rr_file(RRGraphBuilder* rr_graph_builder,
             read_rr_graph_name);
     }
 }
+
+void load_rr_edge_attribute_offset_file(RRGraphBuilder& rr_graph_builder,
+                                        const RRGraphView& rr_graph,
+                                        std::string_view rr_edge_attribute_offset_filename) {
+    std::ifstream file(rr_edge_attribute_offset_filename.data());
+
+    if (!file) {
+        VTR_LOG_ERROR("Failed to open RR edge offset file: %s ", rr_edge_attribute_offset_filename.data());
+    }
+
+    std::map<t_rr_switch_offset_inf, RRSwitchOffsetInfoId> unique_edge_offsets;
+    std::string line;
+    bool first_line = true;
+    while (std::getline(file, line)) {
+        // Ignore first line if it starts with #
+        if (first_line && !line.empty() && line[0] == '#') {
+            first_line = false;
+            continue;
+        }
+        first_line = false;
+
+        std::istringstream iss(line);
+        int edge_id;
+        t_rr_switch_offset_inf rr_switch_detailed_inf;
+
+        if (!(iss >> edge_id >> rr_switch_detailed_inf.Tdel)) {
+            throw std::runtime_error("Invalid line format: " + line);
+        }
+
+        if (edge_id < 0) {
+            throw std::runtime_error("Negative integer found: " + std::to_string(edge_id));
+        }
+
+        auto switch_idx = rr_graph.rr_nodes().edge_switch((RREdgeId)edge_id);
```

Comment:
Good to get some feedback on the file format.  Simple table format?  xml?  binary or xml (capnproto)?  Should store delays, not offsets I think.